src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java

```diff
@@ -404,6 +410,34 @@ public boolean useVariantAnnotations() {
         return Collections.emptyList();
     }
 
+    // TODO: 7/3/18 the two functions below are copy-pasted from GATKTool, and probably some refactoring can be done
```

Comment:
@SHuang-Broad Thanks for doing this. You're right that we definitely don't want to duplicate all of this code. I would factor out a `makeDefaultToolVCFHeaderLines` method that takes the toolkit short name, toolkit version, the tool class name, and the command line, and returns the set of header lines, and delegate to that from both GATKTool and GATKSparkTool when `addOutputVCFCommandLine` is true. For now it can probably live in GATKVariantContextUtils with createVCFWriter. (@droazen @lbergelson we might want to create a ToolUtils class with static stuff like this - I think there is a bunch of code we could move into that).

Reply:
Thanks @cmnbroad .

I looked at `GATKVariantContextUtils.createVCFWriter(...)`, and where  `GATKTool.getDefaultToolVCFHeaderLines(...)` is used, it seems that the individual tools are 
1. constructing their headers by calling explicitly into `GATKTool.getDefaultToolVCFHeaderLines(...)` , then
2. get the writer by calling into `GATKVariantContextUtils.createVCFWriter(...)`, then
3. let the writer `writeHeader()`

in order. So I'm not quite understanding the comment:
>  For now it can probably live in GATKVariantContextUtils with createVCFWriter.

Are you suggesting that I factor out the method `makeDefaultToolVCFHeaderLines(...)`, place it in `GATKVariantContextUtils`, but not let the code be absorbed into `createVCFWriter`?

Reply:
In addition, I'm suspecting more could be done, probably extracting out an abstract class like `AbstractGATKBaseTool`?